crates/pipeline-manager/src/api/endpoints/pipeline_interaction/support_bundle.rs

```diff
@@ -0,0 +1,475 @@
+//! Endpoint to retrieve a support bundles for pipeline.
+use actix_web::rt::time::timeout;
+use chrono::{DateTime, Utc};
+use feldera_types::error::ErrorResponse;
+use futures_util::StreamExt;
+
+use std::io::Write;
+use std::time::Duration;
+use zip::{CompressionMethod, ZipWriter};
+
+use crate::api::endpoints::config::Configuration;
+use crate::api::error::ApiError;
+use crate::api::examples;
+use crate::api::main::ServerState;
+use crate::db::types::pipeline::ExtendedPipelineDescr;
+use crate::db::{storage::Storage, types::tenant::TenantId};
+use crate::error::ManagerError;
+use actix_web::{
+    get,
+    http::Method,
+    web::{self, Data as WebData, ReqData},
+    HttpResponse,
+};
+
+type BundleResult<T> = Result<T, String>;
+
+/// Fetch data from a pipeline endpoint
+async fn fetch_pipeline_data(
+    state: &ServerState,
+    client: &awc::Client,
+    tenant_id: TenantId,
+    pipeline_name: &str,
+    endpoint: &str,
+    timeout_duration: Option<Duration>,
+) -> Result<HttpResponse, ManagerError> {
+    state
+        .runner
+        .forward_http_request_to_pipeline_by_name(
+            client,
+            tenant_id,
+            pipeline_name,
+            Method::GET,
+            endpoint,
+            "",
+            timeout_duration,
+        )
+        .await
+}
+
+/// Stream logs from the pipeline with timeout-based termination
+async fn collect_pipeline_logs(
+    state: &ServerState,
+    client: &awc::Client,
+    tenant_id: TenantId,
+    pipeline_name: &str,
+) -> Result<String, ManagerError> {
+    let mut first_line = true;
+    let next_line_timeout = Duration::from_millis(250);
+    let mut logs = String::with_capacity(4096);
+
+    let response = state
+        .runner
+        .get_logs_from_pipeline(client, tenant_id, pipeline_name)
+        .await?;
+
+    let mut response = response;
+    while let Ok(Some(chunk)) = timeout(
+        if first_line {
+            Duration::from_secs(5)
```

Comment:
Let's be more generous here. Since we anyway wait for 120s for the circuit profile, it's probably ok to wait for 20 seconds here.